Geometry_Engine/Compute/BooleanIntersection.cs

```diff
@@ -298,5 +350,150 @@ public static List<PolyCurve> BooleanIntersection(this List<PolyCurve> regions,
+            bool regSameDir = false;
+            if (Math.Abs(region.INormal().DotProduct(refRegion.INormal()) - 1) <= tolerance)
+                regSameDir = true;
+
+            for (int i = 0; i < tmpResult.Count; i++)
+            {
+                 for (int j = 0; j < tmpResult.Count; j++)
+                {
+                    if (i != j && tmpResult[i].IsGeometricallyEqual(tmpResult[j], tolerance))
+                    {
+                        bool sameDir = tmpResult[i].ITangentAtParameter(0.5).IsEqual(tmpResult[j].ITangentAtParameter(0.5));
+                        if ((regSameDir && sameDir) || (!regSameDir && !sameDir))
+                        {
+                            tmpResult.RemoveAt(Math.Min(j, i));
+                            if (i > j)
+                                i--;
+                            else
+                                j = 0;
+                        }
+                        else
+                        {
+                            tmpResult.RemoveAt(Math.Max(j, i));
+                            tmpResult.RemoveAt(Math.Min(j, i));
+                            if (i > 0)
+                                i--;
+                            else
+                                j = 0;
+                        }
+                    }                        
+                }
+            }
+
+            List<ICurve> result = IJoin(tmpResult, tolerance).Cast<ICurve>().ToList();
+
+            int res = 0;
+            while (res < result.Count)
+            {
+                if (result[res].IArea() <= sqTol || !result[res].IIsClosed(tolerance))
+                    result.RemoveAt(res);
+                else
+                    res++;
+            }
+
+            return result;
+        }
+
+        /***************************************************/
+
+        public static List<ICurve> IBooleanIntersection(this List<ICurve> regions, double tolerance = Tolerance.Distance)
+        {
+            if (regions.Count < 2)
+            {
+                return regions;
+            }
+
+            List<ICurve> result = new List<ICurve>();
+
+            foreach (ICurve region in regions)
+            {
+                if (region.IArea() <= tolerance)
+                    return result;
+            }
+
+            result.Add(regions[0]);
+            for (int i = 1; i < regions.Count; i++)
+            {
+                List<ICurve> newResult = new List<ICurve>();
+                result.ForEach(r => newResult.AddRange(r.IBooleanIntersection(regions[i], tolerance)));
+                result = newResult;
+            }
+
+            return result;
+        }
+
+        /***************************************************/
+        /***          Private methods                    ***/
+        /***************************************************/
+        
+        private static Boolean IsGeometricallyEqual(this ICurve curve, ICurve refCurve, double tolerance = Tolerance.Distance)
```

Comment:
https://github.com/BHoM/BHoM_Engine/blob/master/Environment_Engine/Query/IsIdentical.cs

Take a look at that method, that's how we decided to check for identical geometry on Environment `Panel` - might be worth stealing into here and then I can modify that method to just call this one?

Reply:
I get your point. I need this method to cull (or sometimes not) overlapping edges. At first I was using IsEqual but it was returning nulls for elements of different types what was not acceptable. The possible inputs of my method are segments of regions' edges, split by their intersection points. There is no way in which they could intersect so it is enough to check only 3 points. It as well works for segments being PolyCurve and PolyLines.
I can change it into your way, but now I think it is as light as possible and the methods are already heavy.

Reply:
In that case I would argue the method might not be appropriately named - the way I interpreted it was a check that 2 curves are equal, and those 2 curves can be any length, any shape, etc., so a more definitive check of each point should be made.
Of course there is also a philosophical question about equality - if we have two straight lines for example,  both start and end in the same spot, but one of the lines has an extra middle point geometrically (for whatever reason), can they be considered equal? Of course with the method @pawelbaran and I propose, they would be considered not-equal, because one line has 3 points to the other line which has 2.
Of course that's where a combination of methods can then be used, like the `CleanPolyline` method I put in a few weeks back which would remove that extra point and allow the two lines to be geometrically equal on the discussed methods.

Anyway, I digress...

TL;DR - rename the method if the method is not actually checking geometrical equality - I think that's the summation of that :wink: @pawelbaran would you agree?

Reply:
Ok ok ok ok.. What I meant is that this method checks geometrical equality but **only** in case of my specific method. Because of how the input is created it's elements can't intersect, so if startpoint, midpoint and endpoint are equal that means that elements are equal. 😄  I should have hidden it somewhere in the code so there would be no problem 😄 
Renaming in progres 😉

Reply:
Yeah, agreed with @FraserGreenroyd. My suggestion would be to name it with `IsSimilarSegment` etc.